cpp/transitive-dependencies/build.gradle

```diff
@@ -12,4 +12,22 @@ allprojects {
             }
         }
     }
+
+    task assembleDebuggable {
+        group = "Build"
+        description = "Assemble all debuggable C++ binaries."
+        components.withType(CppExecutable) { binary ->
+            if (binary.isDebuggable()) {
+               dependsOn binary.executableFile
+            }
+        }
+        components.withType(CppSharedLibrary) { binary ->
+            if (binary.isDebuggable()) {
+               dependsOn binary.linkFile
+            }
```

Comment:
This is a bit verbose. We should consider adding an issue to unify the output file under an extra common property so we can do something like:
```
components.withType(CppBinary).matching {it.debuggable}.all { binary ->
  dependsOn binary.outputFile
}
```